src/NSwag.SwaggerGeneration.AspNetCore/AspNetCoreToSwaggerGenerator.cs

```diff
@@ -0,0 +1,257 @@
+                .Any(a => a.GetType().Name == "SwaggerIgnoreAttribute");
+
+            if (hasIgnoreAttribute)
+            {
+                return;
+            }
+
+            var operations = new List<Tuple<SwaggerOperationDescription, ApiDescription, MethodInfo>>();
+
+            var descriptions = new List<SwaggerOperationDescription>();
+            foreach (var item in controllerApiDescriptionGroup)
+            {
+                var apiDescription = item.Item1;
+                var method = item.Item2;
+
+                var path = apiDescription.RelativePath;
+                if (!path.StartsWith("/"))
+                    path = "/" + path;
+
+                var operationDescription = new SwaggerOperationDescription
+                {
+                    Path = path,
+                    Method = GetSwaggerOperationMethod(apiDescription.HttpMethod),
+                    Operation = new SwaggerOperation
+                    {
+                        IsDeprecated = method.GetCustomAttribute<ObsoleteAttribute>() != null,
+                        OperationId = GetOperationId(document, controllerType.Name, method)
+                    }
+                };
+
+                operations.Add(new Tuple<SwaggerOperationDescription, ApiDescription, MethodInfo>(operationDescription, apiDescription, method));
+            }
+
+            await AddOperationDescriptionsToDocumentAsync(document, controllerType, operations, swaggerGenerator, schemaResolver).ConfigureAwait(false);
+        }
+
+        private async Task AddOperationDescriptionsToDocumentAsync(SwaggerDocument document, Type controllerType, List<Tuple<SwaggerOperationDescription, ApiDescription, MethodInfo>> operations, SwaggerGenerator swaggerGenerator, SwaggerSchemaResolver schemaResolver)
+        {
+            var allOperation = operations.Select(t => t.Item1).ToList();
+            foreach (var tuple in operations)
+            {
+                var operation = tuple.Item1;
+                var apiDescription = tuple.Item2;
+                var method = tuple.Item3;
+
+                var addOperation = await RunOperationProcessorsAsync(document, apiDescription, controllerType, method, operation, allOperation, swaggerGenerator, schemaResolver).ConfigureAwait(false);
+                if (addOperation)
+                {
+                    var path = operation.Path.Replace("//", "/");
+
+                    if (!document.Paths.ContainsKey(path))
+                        document.Paths[path] = new SwaggerOperations();
+
+                    if (document.Paths[path].ContainsKey(operation.Method))
+                    {
+                        throw new InvalidOperationException("The method '" + operation.Method + "' on path '" + path + "' is registered multiple times " +
+                            "(check the DefaultUrlTemplate setting [default for Web API: 'api/{controller}/{id}'; for MVC projects: '{controller}/{action}/{id?}']).");
```

Comment:
Is this error message relevant? We don't support ApiExplorer for conventional routes. You can still have duplicates, but the way to fix it is wrong here. Consider a message similar to what we do here https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Core/Internal/ActionSelector.cs#L140